### ISSUE TITLE
Underscores handled differently in operationId and function name by node.js code generator.

### ISSUE THREAD

**Issue by nickform** [swagger-api/swagger-codegen#2302]
Using the app at editor.swagger.io, I generated some node.js code for an api with a path parameter whose name contained an underscore. In the output the operationId generated for this route (in swagger.yaml), the underscore was left in place however in the corresponding function name (in Default.js), the underscore was removed and the following character capitalised. This led to an error when requests were received for the matching url.

**Comment by wing328** [comment]
@nickform thanks for reporting the issue. Please provide the swagger spec for reproducing the issue.

I assume you're generating the `nodejs` (nodejs server stub)

**Comment by nickform** [comment]
[swagger-codegen-issue2302repro.txt](https://github.com/swagger-api/swagger-codegen/files/157263/swagger-codegen-issue2302repro.txt) reproduces the issue (with the train_code parameter.

You are correct that it is node.js server.

**Comment by wing328** [comment]
Using the spec you provided, I found the following:
```
$ grep -R -i traincode *           
controllers/Default.js:module.exports.routeTrainsTrainCodeGET = function routeTrainsTrainCodeGET (req, res, next) {
controllers/Default.js:  Default.routeTrainsTrainCodeGET(req.swagger.params, res, next);
controllers/DefaultService.js:exports.routeTrainsTrainCodeGET = function(args, res, next) {
controllers/DefaultService.js:  * trainCode (String)
```
I can update DefaultServices.js (last line in the above) to use `train_code` instead of `trainCode` but that's just a comment.

Am I supposed to update the following line in Default.js instead?
```
controllers/Default.js:  Default.routeTrainsTrainCodeGET(req.swagger.params, res, next);
```
or can you share a bit more details on what the fixed code looks like?

(disclaimer: not an expert in nodejs server)

**Comment by nickform** [comment]
If I generate Node.js server code from the contents of [swagger-codegen-issue2302repro.txt](https://github.com/swagger-api/swagger-codegen/files/157263/swagger-codegen-issue2302repro.txt) using the app at [editor.swagger.io](editor.swagger.io) I get the following on line 10 of nodejs-server\api\swagger.yaml:

`      operationId: "routeTrainsTrain_codeGET"`

whilst on line 9 of nodejs-server\controllers\Default.js I get:

`module.exports.routeTrainsTrainCodeGET = function routeTrainsTrainCodeGET (req, res, next) {`

(which is the first match you got from your grep command). 

From what I recall when I actually debugged this, the code responsible for routing the request to the right method relied on the operationId being the same as the exported method name which they aren't here.

When I run the code as generated and try to GET using any values for the two parameters I get a 500 response. If I change the operationId on line 10 of the swagger.yaml to:

`      operationId: "routeTrainsTrainCodeGET"`

and restart the app, I get a 200 response. Hope that helps.

**Comment by wing328** [comment]
@nickform I've filed https://github.com/swagger-api/swagger-codegen/pull/2466 to fix the issue.

Would you please perform a test to see if the PR fixes the issue for you?

Here is grep result:
```
swagger-codegen-issue2302repro|⇒ grep -n -R routeTrains  *
api/swagger.yaml:10:      operationId: "routeTrainsTrain_codeGET"
controllers/Default.js:9:module.exports.routeTrainsTrain_codeGET = function routeTrainsTrainCodeGET (req, res, next) {
controllers/Default.js:10:  Default.routeTrainsTrainCodeGET(req.swagger.params, res, next);
controllers/DefaultService.js:3:exports.routeTrainsTrainCodeGET = function(args, res, next) {
swagger-codegen-issue2302repro|⇒

**Comment by nickform** [comment]
I don't know how to build or use swagger-codegen (I just used the one at editor.swagger.io) and I don't have time to learn I'm afraid. If you can provide me with the equivalent of the output I got from the web app but generated with your change, I will be happy to run it and check whether the error has been cleared. Thanks.

**Comment by wing328** [comment]
@nickform I pushed the update code (based on the spec you provided) to https://github.com/wing328/nodejs_server_test.

Would you please take a look?

**Comment by nickform** [comment]
@wing328 I cloned [](https://github.com/wing328/nodejs_server_test) and ran it and can confirm that it does not have the problem I originally reported. For the record I am running node v0.10.25 on Ubunutu. Thanks for fixing. Will you close this issue once your pull request is merged into main?

**Comment by wing328** [comment]
@nickform thanks for the confirmation. We'll merge it into master later as v2.1.6 will be released very soon.

Once it's merged into master, this issue will be closed.

**Comment by wing328** [comment]
@nickform PR merged into master. Please pull the latest and let us know if you still encounter the issue.
